src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -247,19 +231,72 @@ public ExpressionConverter CreateConverterForIndexReplacement(SyntaxBase nameSyn
             return GetConverter(indexContext);
         }
 
-        private LanguageExpression ConvertResourcePropertyAccess(ResourceReferenceExpression reference, PropertyAccessExpression propertyAccess)
+        private LanguageExpression ConvertAccessChain(AccessChainExpression expression)
+        {
+            var (@base, properties, safeAccess) = ProcessAccessExpression(expression.FirstLink);
+            properties = properties.Concat(expression.AdditionalProperties.Select(ConvertExpression));
+
+            return safeAccess
+                ? CreateFunction("tryGet", @base.AsEnumerable().Concat(properties))
+                : AppendProperties(ToFunctionExpression(@base), properties);
+        }
+
+        private LanguageExpression ConvertAccessExpression(AccessExpression expression)
+        {
+            var (@base, properties, safeAccess) = ProcessAccessExpression(expression);
+
+            if (!properties.Any())
+            {
+                return @base;
+            }
+
+            return safeAccess
+                ? CreateFunction("tryGet", @base.AsEnumerable().Concat(properties))
+                : AppendProperties(ToFunctionExpression(@base), properties);
+        }
+
+        private (LanguageExpression @base, IEnumerable<LanguageExpression> properties, bool safeAccess) ProcessAccessExpression(AccessExpression expression)
+        {
+            var (@base, includesFirstAccess) = ConvertBaseExpression(expression);
+
+            if (includesFirstAccess)
+            {
+                return (@base, Enumerable.Empty<LanguageExpression>(), false);
+            }
+
+            var properties = ConvertExpression(expression.Access).AsEnumerable();
+            if (expression is ModuleOutputPropertyAccessExpression)
+            {
+                properties = properties.Append(new JTokenExpression("value"));
+            }
+
+            return (@base, properties, expression.Flags.HasFlag(AccessExpressionFlags.SafeAccess));
+        }
+
+        private (LanguageExpression @base, bool includesFirstAccess) ConvertBaseExpression(AccessExpression expression) => expression.Base switch
+        {
+            ResourceReferenceExpression resource when expression is PropertyAccessExpression exp => GetConverter(resource.IndexContext).ConvertResourcePropertyAccess(resource, exp),
+            ModuleReferenceExpression module when expression is PropertyAccessExpression exp => exp.PropertyName switch
```

Comment:
Worth moving this logic into a `ConvertModulePropertyAccess` (equivalent to `ConvertResourcePropertyAccess`) to simplify this function and demonstrate the similarity between how we handle resource & module properties?